backend/tessend/tessend.go

```diff
@@ -0,0 +1,829 @@
+// It returns a boolean value indicating if tessen should be enabled or not.
+func (t *Tessend) enabled() bool {
+	if !t.config.OptOut {
+		logger.WithField("opt-out", t.config.OptOut).Info("tessen is opted in, enabling tessen.. thank you so much for your support 💚")
+		return true
+	}
+	if t.AllowOptOut {
+		logger.WithField("opt-out", t.config.OptOut).Info("tessen is opted out, patiently waiting for you to opt back in")
+		return false
+	}
+	logger.WithField("opt-out", t.config.OptOut).Info("tessen is opted out but per the license agreement, we're enabling tessen.. thank you so much for your support 💚")
+	return true
+}
+
+// collectAndSend is a durable function to collect and send data to tessen.
+// Errors are logged and tessen continues to the best of its ability.
+func (t *Tessend) collectAndSend() {
+	// collect data
+	data := t.getDataPayload()
+	if err := t.getPerResourceMetrics(time.Now().Unix(), data); err != nil {
+		if err, ok := err.(*store.ErrInternal); ok {
+			select {
+			case t.errChan <- err:
+			case <-t.ctx.Done():
+			}
+		}
+		return
+	}
+
+	logger.WithFields(logrus.Fields{
+		"url":           t.url,
+		"id":            data.Cluster.ID,
+		"metric_points": len(data.Metrics.Points),
+	}).Info("sending resource counts to tessen")
+
+	// send data
+	respHeader := t.send(data)
+	if respHeader == "" {
+		logger.Debug("no tessen response header")
+		return
+	}
+
+	// parse the response header for an integer value
+	interval, err := strconv.ParseUint(respHeader, 10, 32)
+	if err != nil {
+		logger.Debugf("invalid tessen response header: %v", err)
+		return
+	}
+
+	// validate the returned interval is within the upper/lower bound limits
+	err = corev2.ValidateInterval(uint32(interval))
+	if err != nil {
+		logger.Debugf("invalid tessen response header: %v", err)
+		return
+	}
+
+	// update the tessen interval if the response header returns a new value
+	if t.interval != uint32(interval) {
+		t.interval = uint32(interval)
+		logger.WithField("interval", t.interval).Debug("tessen interval updated")
+		t.interrupt <- t.config
+	}
+}
+
+// getDataPayload retrieves cluster, version, and license information
+// and returns the populated data payload.
+func (t *Tessend) getDataPayload() *Data {
+	// collect license information
+
+	// populate data payload
+	data := &Data{
+		Cluster: Cluster{
+			ID:           t.clusterID,
+			Distribution: Distribution,
+			Version:      version.Semver(),
+			License:      t.licenseGetter.Get(),
+		},
+	}
+
+	return data
+}
+
+// getPerResourceMetrics populates the data payload with the total number of each resource.
+func (t *Tessend) getPerResourceMetrics(now int64, data *Data) error {
+	var backendCount float64
+
+	// collect backend count
+
+	// TODO: This needs to be replaced with something else, that maybe we don't
+	// even have yet. Eric mentioned using the Operator Presence Control data,
+	// for which we don't really have an abstraction yet.
```

Comment:
Christian added support for listing OPC items recently, so we can incorporate that now